pallets/collator-selection/src/lib.rs

```diff
@@ -365,7 +375,7 @@ pub mod pallet {
 			let new_candidates = candidates.into_iter().filter_map(|c| {
 				let last_block = <LastAuthoredBlock<T>>::get(c.who.clone());
 				let since_last = now.saturating_sub(last_block);
-				if since_last < kick_threshold {
+				if since_last < kick_threshold || Self::candidates().len() <= T::MinCandidates::get() {
```

Comment:
```suggestion
				if since_last < kick_threshold || Self::candidates().len() <= T::MinCandidates::get() {
```

Why are you reading candidates here but also getting candidates from the input params of this function?

Reply:
so the candidates input does not get changed, it gets at the end collected into new_candidates, so if 5 people get removed, it does not know that, however the global candidates is tracking this, so it would for each candidate have the current knowledge of how many are left after the previous have been kicked